### PR TITLE
Remove cols/rows from internal term.options

### PR DESCRIPTION
**Pull request by silamon**
This branch helps towards #3948. Keeping terminal.rows/terminal.cols and terminal.options.rows/terminal.options.cols in sync shouldn't be a problem then. 

Also aligns xterm with xterm-headless after #3960.

### CODE REVIEW

File: src/browser/Terminal.test.ts [hunk 1]
```diff
@@ -1399,20 +1399,4 @@ describe('Terminal', () => {
...
-      assert.equal(term.options.cols, 80);
-      assert.equal(term.options.rows, 24);
-    });
-    it('set options', async () => {
-      term.options.cols = 40;
-      assert.equal(term.options.cols, 40);
-      term.options.rows = 20;
-      assert.equal(term.options.rows, 20);
-    });
-  });
```
Comment (Tyriar) [inline src/browser/Terminal.test.ts:1417]: We'll still want to keep these tests, just change to use different options

File: src/common/services/BufferService.ts [hunk 2]
```diff
@@ -36,8 +36,8 @@ export class BufferService extends Disposable implements IBufferService {
     @IOptionsService private _optionsService: IOptionsService
   ) {
     super();
-    this.cols = Math.max(_optionsService.rawOptions.cols || 0, MINIMUM_COLS);
-    this.rows = Math.max(_optionsService.rawOptions.rows || 0, MINIMUM_ROWS);
+    this.cols = Math.max(80 || 0, MINIMUM_COLS);
+    this.rows = Math.max(24 || 0, MINIMUM_ROWS);
```
Comment (Tyriar) [inline src/common/services/BufferService.ts:40]: Do the initial dimensions need to be passed into the ctor now?

**Review by Tyriar — CHANGES_REQUESTED** [review]
> Also aligns xterm with xterm-headless after https://github.com/xtermjs/xterm.js/pull/3960.

Opps 🤦

**Comment by silamon** [comment]
Removing cols/rows from ITerminalOptions isn't going to work. Closing this PR and we will have to keep rows and cols in the internal ITerminalOptions.

**Pull request closed without merging**